lib/nostrum/shard/dispatch.ex

```diff
@@ -194,7 +194,7 @@ defmodule Nostrum.Shard.Dispatch do
     do: {event, UserCache.update(p), state}
 
   def handle_event(:VOICE_STATE_UPDATE = event, p, state),
-    do: {event, p, state}
+    do: {event, GuildServer.voice_state_update(p.guild_id, p), state}
```

Comment:
Documentation is one of reasons for structs. They are currently maps, and since this is being implemented properly I'd like them to be in structs. To implement this you simply need to add similar logic to the methods [here](https://github.com/Kraigie/nostrum/blob/08ea92bc4a35d1824f8a92a949898d5557b585e0/lib/nostrum/struct/guild.ex#L138) for the new struct you create.